### PR TITLE
Ai no timers

### PR DESCRIPTION
**Pull request by lulebe**
Removed timers from AI code.

### CODE REVIEW

**Comment by lulebe** [comment]
BTW, only the last 2 commits are relevant. Everything else is the same as initial_ai_integration. Not sure if Github understands that as soon as you review and merge that one, this one will only be 2 commits ahead of master. But anyway, just ignore everything before that.

File: static/js/gamedata/ai.js [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
+    turnResult = makeTurn()
+    gameCanvas.drawGame()
+    switch (turnResult) {
+      case AI_STEP_TYPE.MOVE:
+        await timer(750)
+        break
+      case AI_STEP_TYPE.ATTACK:
+        await timer(4000)
+    }
+  } while (turnResult > 0)
```
Comment (Simsso) [inline static/js/gamedata/ai.js:16]: Add comment "first do-while loop".

File: static/js/gamedata/legacy/ai.js [hunk 2]
```diff
@@ -1,16 +1,28 @@
...
 
+export const AI_STEP_TYPE = {
+   END_TURN: 0,
+   NOTHING: 1,
+   RECRUIT: 2,
+   MOVE: 3,
+   ATTACK: 4
+ }
+
 export function startLegacyAi (game, endTurnCallback) {
```
Comment (Simsso) [inline static/js/gamedata/legacy/ai.js:16]: `endTurnCallback` unused?

File: static/js/gamedata/ai.js [hunk 3]
```diff
@@ -1,12 +1,25 @@
-import { startLegacyAi } from './legacy/ai.js'
+import { startLegacyAi, AI_STEP_TYPE } from './legacy/ai.js'
 
 export default async function (game, gameCanvas) {
+  const makeTurn = startLegacyAi(game)
```
Comment (Simsso) [inline static/js/gamedata/ai.js:4]: `executeSubTurn`?

File: static/js/gamedata/legacy/ai.js [hunk 4]
```diff
@@ -914,241 +937,228 @@ function AiSystemDestroy() {
...
-            console.log("unit found" + unit.type);
-            const _loc3_ = compat.World.getCity(unit.row, unit.col);
-            if (_loc3_ != null) {
-               if (compat.World.CitiesIsProductionFacility(_loc3_)) {
-                  if (player.getGold() > compat.World.DataUnitsGetPrice(DataUnit_ProfileId_Infantry)) {
-                     AiToolsBattle(unit);
-                     // TODO: move to execute step (long term)
-                     DisplayBattleINT = setTimeout(AiSystemExecute_FightOrMoveUnitsStepDone, 4000, player, unit);
-                     return;
+      return false;
```
Comment (Simsso) [inline static/js/gamedata/legacy/ai.js:1109]: Return didFight?

File: static/js/gamedata/legacy/ai.js [hunk 5]
```diff
@@ -914,241 +937,228 @@ function AiSystemDestroy() {
...
-         DisplayBattleINT = setTimeout(AiSystemExecute_StandardUnitsAttackDone, 4000, _loc3_, score, range, unit);
-         return;
+function AiSystemExecute_StandardUnitsWait(player, score, range) {
+   StandardUnitMoving = true;
+   const unit = compat.World.unitList[UnitCounter];
+   if (unit.getState() != Unit_State_Finished) {
+      console.log("attack");
+      let didFight = AiToolsBattle(unit);
+      if (unit.hp) {
+         UnitCounter++;
```
Comment (Simsso) [inline static/js/gamedata/legacy/ai.js]: What if the enemy unit died? Would we skip one?

Comment (Simsso) [reply static/js/gamedata/legacy/ai.js]: Or have index out of bounds in `const unit = compat.World.unitList[UnitCounter];`

**Review by Simsso — CHANGES_REQUESTED** [review]
One counter increment to check, other than that slim and nice upgrade! Especially the long switch deserves a laudatio!

**Review by Simsso — APPROVED** [review]

File: static/js/gamedata/legacy/ai.js (hunk 4 above)
Comment (Simsso) [reply static/js/gamedata/legacy/ai.js:1109]: Nope.

**Pull request merged**